adapters/httpapi_compact.c

```diff
@@ -992,7 +1004,7 @@ static HTTPAPI_RESULT ReceiveHeaderFromXIO(HTTP_HANDLE_DATA* http_instance, unsi
         result = HTTPAPI_READ_DATA_FAILED;
     }
     //Parse HTTP response
-    else if (ParseHttpResponse(buf, &ret) != 1)
+    else if (ParseHttpResponse(buf, &ret, reason, TEMP_BUFFER_SIZE) != 1)
```

Comment:
Why not directly pass in reasonPhrase & maxReasonPhraseSize here?

Reply:
ParseHttpResponse will return error if takes null as parameter